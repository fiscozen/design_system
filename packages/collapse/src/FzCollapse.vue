<script lang="ts" setup>
import { FzCollapseProps } from './types'
import { FzIcon } from '@fiscozen/icons'

defineProps<FzCollapseProps>()
const open = defineModel<boolean>('open')

const handleSummaryClick = () => {
  open.value = !open.value
}
</script>

<template>
  <details :open>
    <summary
      @click.prevent="handleSummaryClick"
      data-e2e-summary
      :class="summaryClass"
      class="text-grey-500 flex h-32 cursor-pointer list-none items-center text-sm"
    >
      <slot name="summary">{{ summary }}</slot>
      <slot name="icon">
        <fz-icon :name="open ? 'chevron-up' : 'chevron-down'" class="ml-32" :size="'md'"></fz-icon>
      </slot>
    </summary>
    <Transition>
      <div data-e2e-content v-show="open" class="text-sm">
        <slot name="content">{{ content }}</slot>
      </div>
    </Transition>
  </details>
</template>

<style scoped>
.v-enter-active,
.v-leave-active {
  transition: opacity 0.5s ease;
}

.v-enter-from,
.v-leave-to {
  opacity: 0;
}
</style>
