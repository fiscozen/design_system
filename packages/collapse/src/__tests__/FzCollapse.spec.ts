import { describe, it, expect } from 'vitest'

import { flushPromises, mount } from '@vue/test-utils'
import FzCollapse from '../FzCollapse.vue'

describe('FzCollapse', () => {
  it('should match snapshot', () => {
    const wrapper = mount(FzCollapse, {
      props: {
        summary: 'this is a test summary',
        content: 'this is test content',
        open: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should open when the summary is clicked', async () => {
    const wrapper = mount(FzCollapse, {
      props: {
        summary: 'this is a test summary',
        content: 'this is test content',
        open: false
      }
    })

    await wrapper.find('[data-e2e-summary]').trigger('click')
    expect(wrapper.find('[data-e2e-content]').isVisible()).toBe(true)
  })
})
