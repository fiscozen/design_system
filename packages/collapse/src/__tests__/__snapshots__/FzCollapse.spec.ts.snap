// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzCollapse > should match snapshot 1`] = `
"<details data-v-d76ce47a="" open="">
  <summary data-v-d76ce47a="" data-e2e-summary="" class="text-grey-500 flex h-32 cursor-pointer list-none items-center text-sm">this is a test summary<svg data-v-d76ce47a="" class="svg-inline--fa fa-chevron-up ml-32" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="chevron-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
      <path class="" fill="currentColor" d="M256 105.4l11.3 11.3 192 192L470.6 320 448 342.6l-11.3-11.3L256 150.6 75.3 331.3 64 342.6 41.4 320l11.3-11.3 192-192L256 105.4z"></path>
    </svg></summary>
  <transition-stub data-v-d76ce47a="" appear="false" persisted="true" css="true">
    <div data-v-d76ce47a="" data-e2e-content="" class="text-sm">this is test content</div>
  </transition-stub>
</details>"
`;
